### PR TITLE
generate unique ids when mocking nanoid

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- generate unique ids when mocking nanoid

### CODE REVIEW

File: ui/src/setupTests.ts [hunk 1]
```diff
@@ -6,12 +6,4 @@ import { enableFetchMocks } from "jest-fetch-mock";
 
 Enzyme.configure({ adapter: new Adapter() });
 
-jest.mock("@reduxjs/toolkit", () => {
-  const toolkit = jest.requireActual("@reduxjs/toolkit");
-  return {
-    ...toolkit,
-    nanoid: jest.fn(() => "Uakgb_J5m9g-0JDMbcJqLJ"),
```
Comment (huwshimi) [inline ui/src/setupTests.ts:13]: This was previously returning `undefined` as it was being reset between tests.

File: ui/src/app/base/hooks/base.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { useCallback, useEffect, useRef, useState } from "react";
 
 import { NotificationSeverity } from "@canonical/react-components";
 import type { NotificationProps } from "@canonical/react-components";
-import { nanoid } from "nanoid/non-secure";
+import { nanoid } from "@reduxjs/toolkit";
```
Comment (huwshimi) [inline ui/src/app/base/hooks/base.ts:5]: @petermakowski `nanoid` exported from redux toolkit is the nonsecure version: https://redux-toolkit.js.org/api/other-exports#nanoid.

Comment (huwshimi) [reply ui/src/app/base/hooks/base.ts:5]: I meant to also say that by switching it back to the reduxjs/toolkit version it gives us one less thing to mock.

**Review by Caleb-Ellis — APPROVED** [review]

File: ui/src/app/base/hooks/base.ts (hunk 2 above)
Comment (petermakowski) [reply ui/src/app/base/hooks/base.ts:5]: This was done on purpose, hence the comment:
```jsx
 * @returns non-secure random ID string
```
https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/base/hooks/base.ts#L164

non-secure version is faster and we don't really need enhanced security when generating IDs for the DOM elements.

**Pull request merged**